src/Stratis.Features.FederatedPeg/TargetChain/WithdrawalTransactionBuilder.cs

```diff
@@ -50,20 +51,26 @@ public Transaction BuildWithdrawalTransaction(uint256 depositId, uint blockTime,
                 uint256 opReturnData = depositId;
                 string walletPassword = this.federationWalletManager.Secret.WalletPassword;
                 bool sign = (walletPassword ?? "") != "";
-                var multiSigContext = new TransactionBuildContext(new[]
-                {
-                    recipient.WithPaymentReducedByFee(this.federationGatewaySettings.TransactionFee)
-                }.ToList(), opReturnData: opReturnData.ToBytes())
+
+                var multiSigContext = new TransactionBuildContext(new List<Recipient>(), opReturnData: opReturnData.ToBytes())
                 {
-                    TransactionFee = this.federationGatewaySettings.TransactionFee,
                     MinConfirmations = MinConfirmations,
                     Shuffle = false,
                     IgnoreVerify = true,
                     WalletPassword = walletPassword,
                     Sign = sign,
-                    Time = this.network.Consensus.IsProofOfStake ? blockTime : (uint?) null
+                    Time = this.network.Consensus.IsProofOfStake ? blockTime : (uint?) null,
                 };
 
+                multiSigContext.Recipients = new[] { recipient.WithPaymentReducedByFee(FederationGatewaySettings.CrossChainTransferFee) }.ToList(); // The fee known to the user is taken.
```

Comment:
```
// The fee known to the user is taken.
multiSigContext.Recipients = new List<Recipient>() { recipient.WithPaymentReducedByFee(FederationGatewaySettings.CrossChainTransferFee) };
```

to avoid creating list and array on the same line.